hal/armv7a/_interrupts.S

```diff
@@ -19,14 +19,32 @@
 
 .arm
 
+.extern schedulerLocked
+
+.macro push_fpu_state reg_tmp
+	vpush {d16-d31}
+	vpush {d0-d15}
+	vmrs \reg_tmp, fpscr
+	push {\reg_tmp}
+.endm
+
+.macro unlock_scheduler
+	ldr r0, =schedulerLocked
+	mov r1, #0
+	dmb
+	str r1, [r0]
```

Comment:
`strex` perhaps?

Reply:
Exclusive monitors check memory locations for any write operation - so a normal `str` is enough, we don't need to do a `ldrex`, `strex` sequence.